frontend/src/layout/Header.tsx

```diff
@@ -62,85 +64,255 @@ const Search: React.FC = () => {
+                    display: "block" as const,
+                    position: "absolute" as const,
+                    bottom: -arrowSize / 2,
+                    backgroundColor: "white",
+                    zIndex: 10,
+                    width: arrowSize,
+                    height: arrowSize,
+                    borderTop: "1px solid #bbb",
+                    borderLeft: "1px solid #bbb",
+                    transform: "rotate(45deg)",
+
+                    // Horizontally center
+                    left: 0,
+                    right: 0,
+                    margin: "auto",
+                },
+            },
+        }}>
+            <div
+                title={title}
+                onClick={onClick}
+                css={{
+                    padding: 5,
+                    margin: "0 4px",
+                    borderRadius: 4,
+                    lineHeight: 0,
+                    cursor: "pointer",
+                    fontSize: iconSize,
+                    "&:hover": {
+                        backgroundColor: "#ddd",
+                    },
+                }}
+            >{children}</div>
+        </div>
+    );
+};
+
+// The main menu with several links and functions.
+const MainMenu: React.FC<{ closeMenu: () => void }> = ({ closeMenu }) => {
+    const { t } = useTranslation();
+    const iconStyle = {
+        display: "inline-block",
+        marginRight: 16,
+        color: "#666",
+    };
+
+    const itemStyle = {
+        display: "block",
+        whiteSpace: "nowrap" as const,
+        padding: "6px 8px 6px 12px",
+        textDecoration: "none",
+        "&:hover": {
+            backgroundColor: "#ddd",
+        },
+    };
+
+    return <>
+        <div css={{
+            borderBottom: "1px solid #bbb",
+            textAlign: "center",
+            minWidth: 220,
+            padding: 8,
+        }}>
+            {t("login.not-logged-in")}
+            <LoginButton closeMenu={closeMenu}/>
+        </div>
+        <ul css={{
+            width: "100%",
+            listStyle: "none",
+            margin: 0,
+            padding: 0,
+            fontSize: 18,
+        }}>
+            <li css={itemStyle}>
+                <FontAwesomeIcon icon={faMoon} fixedWidth css={iconStyle} />
+                {t("main-menu.theme")}
+            </li>
+            <li>
+                <Link to="/settings" css={itemStyle} onClick={closeMenu}>
+                    <FontAwesomeIcon icon={faCog} fixedWidth css={iconStyle} />
+                    {t("main-menu.settings")}
+                </Link>
+            </li>
+            <li>
+                <Link to="/about" css={itemStyle} onClick={closeMenu}>
+                    <FontAwesomeIcon icon={faInfoCircle} fixedWidth css={iconStyle} />
+                    {t("main-menu.about")}
+                </Link>
+            </li>
+        </ul>
+    </>;
+};
+
+const LoginButton: React.FC<{ closeMenu: () => void }> = ({ closeMenu }) => {
```

Comment:
I would not call this prop `closeMenu`. Why not just `onClick`? The button doesn't need to know that it's in a menu, just that people might be interested about it being clicked.